providers/openshift/provider_test.go

```diff
@@ -93,3 +132,49 @@ func TestDontPassBasicAuthentication(t *testing.T) {
 		t.Errorf("access token should be empty string for basic authentication: %v", session)
 	}
 }
+
+func TestNewOpenShiftClient(t *testing.T) {
```

Comment:
nit check that length of `httpClientCache sync.Map` via `Range` at each point in the test to make sure it goes from `0` to `1` to `2`.